### ISSUE TITLE
False-positive E1129 (not-context-manager) with tensorflow 2.2

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#3596]
### Steps to reproduce
```
"""False-positive not-context-manager."""
import tensorflow

with tensorflow.summary.create_file_writer(".").as_default():
    pass
```

### Current behavior
> Context manager 'generator' doesn't implement __enter__ and __exit__.

### Expected behavior
Perfect score.

### pylint --version output
```
pylint 2.5.2
astroid 2.4.1
Python 3.8.2 (default, Feb 26 2020, 09:52:54) 
[GCC 7.5.0]
```

Als happens in this configuration:
```
pylint 2.5.2
astroid 2.4.1
Python 3.7.7 (default, May  5 2020, 20:41:09) 
[GCC 7.5.0]
```

Problem appears to be specific to TensorFlow 2.2: `tensorflow==2.2.0rc4` has it, `tensorflow==2.1.0` does not.

**Comment by mihaimaruseac** [comment]
Was this issue also opened on TF repo? It seems like a regression

**Comment by bersbersbers** [comment]
> Was this issue also opened on TF repo?

Not by me at least.

**Comment by bersbersbers** [comment]
Cannot reproduce any longer with `tensorflow==2.8.0rc0` and 
```
pylint 2.12.2
astroid 2.9.3
Python 3.9.10 (main, Jan 17 2022, 14:25:44) 
[GCC 7.5.0]
```

I am even getting a `useless-supression` now with this code:

```python
"""False-positive not-context-manager."""
import tensorflow

# pylint:disable=not-context-manager
with tensorflow.summary.create_file_writer(".").as_default():
    pass
```
